### PR TITLE
add stdin_open, tty support, add tests, fix #344

### LINKED ISSUE #344: support for stdin_open and tty keys
Hi,

My docker-compose.yml looks something like - 

```yaml
version: "2"
services:
  client:
     image: registry.centos.org/centos/centos:7
     stdin_open: true
     tty: true
```

`kompose up` gives me -
```bash
WARN[0000] Unsupported stdin_open key - ignoring        
WARN[0000] Unsupported tty key - ignoring               
WARN[0000] [client] Service cannot be created because of missing port. 
We are going to create Kubernetes Deployments, Services and PersistentVolumeClaims for your Dockerized application. 
If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead. 

INFO[0000] Successfully created Deployment: client      

Your application has been deployed to Kubernetes. You can run 'kubectl get deployment,svc,pods,pvc' for details.
```

Since these keys are unsupported, the pod goes from Running to Completed to CrashLoopBackOff

```bash
$ kubectl get pods -w
NAME                      READY     STATUS              RESTARTS   AGE
client-2474090665-o7wdc   0/1       ContainerCreating   0          1s
NAME                      READY     STATUS      RESTARTS   AGE
client-2474090665-o7wdc   0/1       Completed   0          1s
client-2474090665-o7wdc   0/1       Completed   1         2s
client-2474090665-o7wdc   0/1       CrashLoopBackOff   1         3s
client-2474090665-o7wdc   0/1       Completed   2         16s
```

It would be great if `stdin_open` and `tty` could be supported.

Here is the equivalent in Kubernetes world - 

```yaml
apiVersion: extensions/v1beta1
kind: Deployment
...
spec:
...
  template:
    metadata:
    ...
    spec:
      containers:
      - stdin: true
        tty: true
        image: registry.centos.org/centos/centos:7
```

### PR DESCRIPTION
**Pull request by containscafeine**
This adds supports for stdin_open: bool and
tty: bool support for kubernetes and openshift
providers in kompose. This maps to the
template.Spec.Containers[0].Stdin and
template.Spec.Containers[0].TTY in Kubernets
world.

Also, added tests.

Fixes #344

### CODE REVIEW

File: pkg/transformer/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -428,6 +434,8 @@ func (k *Kubernetes) InitPod(name string, service kobject.ServiceConfig) *api.Po
 				{
 					Name:  name,
 					Image: service.Image,
+					Stdin: service.Stdin,
+					TTY:   service.Tty,
```
Comment (surajssd) [inline pkg/transformer/kubernetes/kubernetes.go]: as opposed to doing it in multiple places you can do it in one place https://github.com/kubernetes-incubator/kompose/blob/master/pkg/transformer/kubernetes/k8sutils.go#L327

File: pkg/transformer/openshift/openshift_test.go [hunk 2]
```diff
@@ -60,6 +62,15 @@ func TestInitDeploymentConfig(t *testing.T) {
 	if spec.Spec.Triggers[1].ImageChangeParams.ContainerNames[0] != "myfoobarname" {
 		t.Errorf("Expected myfoobarname for name, actual %s", spec.Spec.Triggers[1].ImageChangeParams.ContainerNames[0])
 	}
+
+	if spec.Spec.Template.Spec.Containers[0].Stdin != true {
```
Comment (surajssd) [inline pkg/transformer/openshift/openshift_test.go]: Can we create a separate test? Because, this test gets hidden and never shows on output as this was tested?

File: pkg/transformer/kubernetes/kubernetes.go (hunk 1 above)
Comment (containscafeine) [inline pkg/transformer/kubernetes/kubernetes.go]: Fixed, thanks!

File: pkg/transformer/openshift/openshift_test.go (hunk 2 above)
Comment (containscafeine) [inline pkg/transformer/openshift/openshift_test.go]: Added another test instead of this!

File: pkg/transformer/openshift/openshift_test.go [hunk 3]
```diff
@@ -44,6 +46,29 @@ func newServiceConfig() kobject.ServiceConfig {
...
 		User:          "user", // not supported
+		Stdin:         true,
+		Tty:           true,
+	}
+}
+
+func TestOpenShiftUpdateKubernetesObjects(t *testing.T) {
+	var object []runtime.Object
+	o := OpenShift{}
+	serviceConfig := newServiceConfig()
```
Comment (surajssd) [inline pkg/transformer/openshift/openshift_test.go]: A log saying what this test runs will help.

Comment (surajssd) [inline pkg/transformer/openshift/openshift_test.go]: Similar comments are missing in other tests!

**Comment by surajssd** [comment]
apart from those two comments LGTM!

File: pkg/transformer/openshift/openshift_test.go (hunk 3 above)
Comment (containscafeine) [inline pkg/transformer/openshift/openshift_test.go]: Fixed for `TestOpenShiftUpdateKubernetesObjects`, log messages for other tests should go in another PR. Tracked at #351

**Comment by surajssd** [comment]
@containscafeine awesome work :+1:
